### PR TITLE
'PEP440 and "version always up to date" tests' after 'Simplified Version and added support for PEP440'

### PR DESCRIPTION
**Pull request by ceball**

### CODE REVIEW

File: scripts/tmpverify [hunk 1]
```diff
@@ -32,14 +32,15 @@ else:
 newv=[int(x) for x in v[1::].split('.')];
 #newv[-1]+=1;
 predicted_version='.'.join(str(x) for x in newv)+'.post'+commits+"+"+sha;
+predicted_version = predicted_version.strip() # Strip off trailing newline
 
-assert autover.__version__==predicted_version, '%s != %s'%(autover.__version__,predicted_version)
+assert autover.__version__.startswith(predicted_version), '%r does not start with %r' % (autover.__version__,predicted_version)
```
Comment (ceball) [inline scripts/tmpverify]: Could you explain this change?

**Comment by jlstevens** [comment]
@ceball The test build is now passing but the 'Conda_package' build is failing with:

```
File "/home/travis/build/ioam/autover/conda.recipe/meta.yaml", line 3, in top-level template code
    version: {{ environ['GIT_DESCRIBE_TAG'].lstrip('v') }}.{{ GIT_BUILD_STR }}
TypeError: 'NoneType' object is not callable
```

Which should have nothing to do with autover, but something to do with the meta.yaml definition/templating.

I made three basic changes:

1. Made sure ``__version__`` is always a string (solution discussed in  #14)
2. Fixed handling of file paths - the ``tmpverify`` script works inside a different current working directory. I switch to using ``__file__`` which fixes this (this was a painful one to debug!)
3. Offered different policies of what to do about a dirty repo in setup.py: ``'report'``, ``'strip'``, ``'raise'``. 
  * The ``'report'`` option is the old policy that reports if dirty. 
  * The ``'strip'`` policy never reports ``'-dirty'`` (what I used in this PR and why I updated ``tmpverify`` to use ``startswith``). 
  * Finally, you can call ``get_setup_version`` with ``dirty='raise'`` if you want to block things from happening when the repo is dirty (e.g when building packages). Obviously you will need to only do this sometimes (e.g for ``setup.py sdist``) and needs autover as a build dependency.

The three policy options should offer the necessary flexibility to handle dirty repositories in setup.py.

**Comment by jlstevens** [comment]
Yay! It went green!

**Comment by jlstevens** [comment]
Yay! All the travis tests are passing at last!

**Comment by jlstevens** [comment]
Just to make sure we don't forget:

- [x] Get appveyor tests passing
- [x] ``ref`` of the embed function will need to be updated to an appropriate git tag.
- [ ] This is needed in setup.py ``    package_data = {'pkg_bundle': ['.version']},`` or alternatively the ``MANIFEST`` (is it either/or...or are both needed?)
- [ ] There is a fallback import in the package ``__init__.py`` files and the JSON fallback needs to be readded.
- [x] We should switch on Python 2 tests in the build matrix.
- [x] We need to decide whether we are renaming ``version.py`` to ``autover.py`` or not (will be handled after merging into the original pep440 PR). 
- [x] Make sure the ``setup.py`` files match in terms of code and only differ by ``reponame``.

I believe our recommendation now is to *discourage* tracking the ``.version`` file with git.

**Comment by jlstevens** [comment]
@ceball I think I would like to merge this PR once python 2 is added to the build matrix and all the tests are green. Then we can tidy things up in other, much smaller PRs. Does that make sense to you?

**Comment by ceball** [comment]
No problem, you are free to merge this any time.

**Comment by ceball** [comment]
(As in, I'll stop now as you requested on flowdock.)

**Comment by jlstevens** [comment]
Great! Very happy to have these tests. Now we just need to enable the python 2 tests for everything and maybe add more unit tests. We could enable mac tests on travis but I see that as optional...

**Comment by ceball** [comment]
I think enabling mac tests would be a bad idea for this project. Unlike windows and linux tests, mac tests will take a long time to start up and run. Meanwhile, the extra coverage provided by running on mac seems like it would be pretty small for this (dependency free) project, unless I'm missing something?

**Comment by jlstevens** [comment]
No, I agree with that analysis! If linux + windows works, mac almost certainly does and it isn't worth introducing those slow tests.

**Pull request merged**